text/trace/0220-messaging-semantic-conventions-span-structure.md

```diff
@@ -0,0 +1,484 @@
+each message it accounts for, the "Deliver" or "Receive" span SHOULD link to
+the "Create" or "Publish" span for the message.
+
+#### Settlement spans
+
+"Settle" spans or events SHOULD be created for every manually or automatically
+triggered settlement operation. A single "Settle" span can account for a
+single message or for multiple messages (in case messages are passed for
+settling as batches). For each message it accounts for, the "Settle" span
+MAY link to the "Create" span for the message.
+
+## Examples
+
+### Single message producer, single message push-based consumer
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  PM1[Publish m1]
+  end
+  subgraph CONSUMER
+  DM1[Deliver m1]
+  end
+  PM1-. link .->DM1;
+```
+
+### Single message producer, single message push-based consumer with manual settlement
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+    direction TB
+    PM1[Publish m1]
+  end
+  subgraph CONSUMER
+    direction TB
+    DM1[Deliver m1]-->S1[Settle m1]
+  end
+  PM1-. link .->DM1;
+```
+
+### Single message producer, single message push-based consumer with auto-settlement
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction TB
+  PM1[Publish m1]
+  end
+  subgraph CONSUMER
+  direction TB
+  A[Ambient]-->DM1[Deliver m1]
+  A-->S1[Settle m1]
+  end
+  PM1-. link .->DM1;
+```
+
+### Batch message producer with "Create" spans, single message pull-based consumer
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction TB
+  P[Publish]-->CM1[Create m1]
+  P-->CM2[Create m2]
+  end
+  subgraph CONSUMER
+  direction TB
+  RM1[Receive m1]
+  RM2[Receive m1]
+  end
+  CM1-. link .->RM1;
+  CM2-. link .->RM2;
+```
+
+### Batch message producer, single message push-based consumer
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction TB
+  P[Publish]
+  end
+  subgraph CONSUMER
+  direction TB
+  DM1[Deliver m1]
+  DM2[Deliver m1]
+  end
+  P-. link .->DM1;
+  P-. link .->DM2;
+```
+
+### Batch message producer with "Create" spans populated before publish, single message pull-based consumer
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction TB
+  A[Ambient]-->CM1[Create m1]
+  A-->CM2[Create m2]
+  A-->P[Publish]
+  end
+  subgraph CONSUMER
+  direction TB
+  RM1[Receive m1]
+  RM2[Receive m1]
+  end
+  CM1-. link .->RM1;
+  CM2-. link .->RM2;
+```
+
+### Single message producers, batch push-based consumer with process spans
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction TB
+  PM1[Publish m1]
+  PM2[Publish m2]
+  end
+  subgraph CONSUMER
+  direction TB
+  D[Deliver]-->PRM1[Process m1]
+  D-->PRM2[Process m2]
+  end
+  PM1-. link .->D;
+  PM2-. link .->D;
+```
+
+### Single message producers, batch pull-based consumer with process spans
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction TB
+  PM1[Publish m1]
+  PM2[Publish m2]
+  end
+  subgraph CONSUMER
+  direction TB
+  A[Ambient]-->R[Receive]
```

Comment:
I'm not sure what ambient means in this case, but I don't believe it's possible to have ambient context here (created by instrumentation) and in other pull-based examples